src/document/operation/changeoperation.js

```diff
@@ -124,6 +97,35 @@ CKEDITOR.define( [ 'document/operation/operation', 'ckeditorerror' ], ( Operatio
 		getReversed() {
 			return new ChangeOperation( this.range, this.newAttr, this.oldAttr, this.baseVersion + 1 );
 		}
+
+		clone() {
+			return new ChangeOperation( this.range.clone(), this.oldAttr, this.newAttr, this.baseVersion );
```

Comment:
You introduced `clone` methods to all operation classes, position class and range class. On the one hand we could use `utils.clone`, but on the other cloned object is not the instance of the class (`( utils.clone( position ) instanceof Position ) === false`) and `utils.clone` will not know how deep the object should be cloned (in this case range have to be cloned, but attributes not), so this is fine for me.